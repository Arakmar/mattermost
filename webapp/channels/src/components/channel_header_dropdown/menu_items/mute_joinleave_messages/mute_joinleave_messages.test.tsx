// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {shallow} from 'enzyme';
import React from 'react';
import {TestHelper} from 'utils/test_helper';

import Menu from 'components/widgets/menu/menu';

import MuteJoinLeaveMessages from './mute_joinleave_messages';

describe('components/ChannelHeaderDropdown/MenuItem.MuteJoinLeaveMessages', () => {
    const baseProps = {
        channel: TestHelper.getChannelMock({
            id: 'channel_id',
            type: 'O',
        }),
        actions: {
            patchChannel: jest.fn(),
        },
    };

    it('should match snapshot', () => {
        const wrapper = shallow<typeof MuteJoinLeaveMessages>(<MuteJoinLeaveMessages {...baseProps}/>);
        expect(wrapper).toMatchSnapshot();
    });

    it('should read "show join/leave messages"', () => {
        const props = {
            ...baseProps,
            channel: {
                ...baseProps.channel,
                options: {
                    excludeTypes: ['system_join_channel', 'system_leave_channel'],
                },
            },
        };
        const wrapper = shallow<typeof MuteJoinLeaveMessages>(<MuteJoinLeaveMessages {...props}/>);
        expect(wrapper).toMatchSnapshot();
    });

    it('should read "hide join/leave messages"', () => {
        const props = {
            ...baseProps,
            channel: {
                ...baseProps.channel,
                options: {
                    excludeTypes: [],
                },
            },
        };
        const wrapper = shallow<typeof MuteJoinLeaveMessages>(<MuteJoinLeaveMessages {...props}/>);
        expect(wrapper).toMatchSnapshot();
    });

    it('should run patchChannel function on click to set as muted', () => {
        const props = {
            ...baseProps,
            channel: {
                ...baseProps.channel,
                options: {
                    excludeTypes: [],
                },
            },
            actions: {...baseProps.actions},
        };
        const wrapper = shallow<typeof MuteJoinLeaveMessages>(<MuteJoinLeaveMessages {...props}/>);

        wrapper.find(Menu.ItemAction).simulate('click', {
            preventDefault: jest.fn(),
        });
        expect(props.actions.patchChannel).toHaveBeenCalledWith(props.channel.id, {
            options: {
                excludeTypes: ['system_join_channel', 'system_leave_channel'],
            },
        });
    });
    it('should run patchChannel function on click to set as unmuted', () => {
        const props = {
            ...baseProps,
            channel: {
                ...baseProps.channel,
                options: {
                    excludeTypes: ['system_join_channel', 'system_leave_channel'],
                },
            },
            actions: {...baseProps.actions},
        };
        const wrapper = shallow<typeof MuteJoinLeaveMessages>(<MuteJoinLeaveMessages {...props}/>);

        wrapper.find(Menu.ItemAction).simulate('click', {
            preventDefault: jest.fn(),
        });
        expect(props.actions.patchChannel).toHaveBeenCalledWith(props.channel.id, {
            options: {
                excludeTypes: [],
            },
        });
    });
    it('should run patchChannel function on click to set as unmuted and persist other exclude types', () => {
        const props = {
            ...baseProps,
            channel: {
                ...baseProps.channel,
                options: {
                    excludeTypes: ['system_join_channel', 'system_leave_channel', 'another_one'],
                },
            },
            actions: {...baseProps.actions},
        };
        const wrapper = shallow<typeof MuteJoinLeaveMessages>(<MuteJoinLeaveMessages {...props}/>);

        wrapper.find(Menu.ItemAction).simulate('click', {
            preventDefault: jest.fn(),
        });
        expect(props.actions.patchChannel).toHaveBeenCalledWith(props.channel.id, {
            options: {
                excludeTypes: ['another_one'],
            },
        });
    });
});
