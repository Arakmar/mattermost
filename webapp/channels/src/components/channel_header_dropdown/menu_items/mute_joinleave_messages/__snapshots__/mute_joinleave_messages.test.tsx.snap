// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ChannelHeaderDropdown/MenuItem.MuteJoinLeaveMessages should match snapshot 1`] = `
<MenuItemAction
  onClick={[Function]}
  show={true}
  text=""
/>
`;

exports[`components/ChannelHeaderDropdown/MenuItem.MuteJoinLeaveMessages should read "hide join/leave messages" 1`] = `
<MenuItemAction
  onClick={[Function]}
  show={true}
  text=""
/>
`;

exports[`components/ChannelHeaderDropdown/MenuItem.MuteJoinLeaveMessages should read "show join/leave messages" 1`] = `
<MenuItemAction
  onClick={[Function]}
  show={true}
  text=""
/>
`;
